cpu: Add a 2nd label 'package' to metric node_cpu_core_throttles_total

knweiss opened the pull request:
This fixes the `node_cpu_core_throttles_total` metrics on multi-socket systems as explained in detail in my comment to PR #836. As the `core_ids` are the same for each node (aka processor or package) we need to count them seperately. Otherwise we only count the last processor's cpu core throttles.

Refactor the `core_throttles_count` handling and move it from the cpu loop
to the NUMA node loop.

Reorganize the `sys.ttar` files and use the same symlinks as the Linux
kernel.

SuperQ commented:
Thanks!  Can you add a fixture example for a second `node`?

grobie commented:
Could we consider not using the term `node` for that label? It already has a prominent meaning in the node exporter.

SuperQ commented on collector/cpu_linux.go:
```diff
@@ -74,7 +74,7 @@ func NewCPUCollector() (Collector, error) {
 			prometheus.BuildFQName(namespace, cpuCollectorSubsystem, "core_throttles_total"),
 			"Number of times this cpu core has been throttled.",
-			[]string{"core"}, nil,
+			[]string{"node", "core"}, nil,
```
How about we call this `package` instead of node.

discordianfish commented:
Besie @SuperQ's comment, LGTM!

SuperQ commented:
@knweiss Thanks for all the research to figure this out.

SuperQ commented:
One additional request, I'd like to start including `CHANGELOG` entries in PRs.  Would you please add a `[BUGFIX]` entry and mention it under `**Breaking changes**`.

knweiss commented:
I've tried to simplify and fix the code. This a first version (without much testing!).

@SuperQ I've added CHANGELOG entries.
@SuperQ I've also added a 2nd (actually a third) node. See the commit message for details.
@grobie The label of the metrics `node_cpu_core_throttles_total` and `node_cpu_package_throttles_total` is now called `package` instead of `node`.

SuperQ commented on CHANGELOG.md:
```diff
@@ -2,10 +2,10 @@
 **Breaking changes**
 
-* [CHANGE]
+* [CHANGE] Rename `node_package_throttle` label `node` to `package`. #871
```
This should be just listed under the breaking changes section, we typically list one item per PR.

mjtrangoni commented:
:+1: LGTM

knweiss commented:
In the meantime I did some more testing on CentOS 6.9 and 7.4 machines with no, partial, and full core+package throttle information in /sys. Both Intel and AMD. So far it works fine and I even caught a couple of machines with a throttle count != 0 (both core and package). YMMV.

knweiss commented:
@SuperQ It would be very useful to have fixtures from all the various cpu types, archs, and kernel versions and an e2e test that tries them all one after the other. This would allow making changes with more confidence...

On the other hand it would, of course, require more effort to keep all the e2e reference files up-to-date.

SuperQ commented:
Yea, I think this is the case where we should move all of this code over to the `procfs` library and have a series of unit tests for various sample fixtures.  This would be much more maintainable than doing it here in the node_exporter.

mjtrangoni commented:
@SuperQ I agree with you. I will be making a PR exporting `/sys/devices/system/cpu/*`. See [#84](https://github.com/prometheus/procfs/issues/84). 
IMHO you could merge this PR, and then in a separate PR use the `procfs` implementation.

discordianfish commented:
Let's merge it, we can use the updated procfs later.
